tests/views/test_cat_view.py

```diff
@@ -204,6 +204,18 @@ def test_list_cats(
     )
 
 
+@mock.patch("myfirstcatapi.domains.cat_domain.get_total_cats_count")
+def test_get_total_cats_count(
+    mock_cat_domain_get_total_cats_count: mock.Mock,
+) -> None:
+    mock_cat_domain_get_total_cats_count.return_value = 1
+
+    response = client.get("/v1/cat/count")
```

Comment:
[suggestion] Easiest way of pagination is not using count but instead start at page 1 and see if you get back `hasNextPage` from the API and if so, show the next link. This way we do not need to care what the count is which makes things faster (otherwise we have to call API twice; once for the count and then again for the actual results)

Reply:
done